llama_index/indices/knowledge_graph/base.py

```diff
@@ -63,21 +75,39 @@ def __init__(
                 max_knowledge_triplets=self.max_triplets_per_chunk
             )
         )
+        self._graph_store = graph_store or SimpleGraphStore()
 
         super().__init__(
             nodes=nodes,
             index_struct=index_struct,
+            service_context=service_context,
+            storage_context=storage_context,
             **kwargs,
         )
 
+        # TODO: legacy conversion - remove in next release
+        if (
+            len(self.index_struct.table) > 0
+            and isinstance(self.graph_store, SimpleGraphStore)
+            and len(self.graph_store._data.graph_dict) == 0
+        ):
+            logger.warning("Upgrading previously saved KG index to new storage format.")
+            self.graph_store._data.graph_dict = self.index_struct.rel_map
+
+    @property
+    def graph_store(self) -> GraphStore:
+        return self._graph_store
+
     def as_retriever(self, **kwargs: Any) -> BaseRetriever:
-        from llama_index.indices.knowledge_graph.retrievers import (
+        from llama_index.indices.knowledge_graph.retriever import (
             KGRetrieverMode,
             KGTableRetriever,
         )
 
         if len(self.index_struct.embedding_dict) > 0 and "retriever_mode" not in kwargs:
             kwargs["retriever_mode"] = KGRetrieverMode.HYBRID
+        if "graph_store" not in kwargs:
```

Comment:
I don't think we need to inject into kwargs. We already have the reference to the index, which exposes the graph store?